products/memgraph/upcoming/tutorials/understanding-music-with-modules.md

```diff
@@ -0,0 +1,450 @@
+        info = {
+            'size': len(community),
+        }
+
+        genre_count = defaultdict(lambda: 0)
+        for genre in itertools.chain(*[u.properties["genres"] for u in community]):
+            genre_count[genre] += 1
+
+        if len(genre_count) != 0:
+            mpg = max(genre_count.items(), key=lambda item: item[1])
+
+            info['most_popular_genre'] = mpg[0]
+            info['most_popular_genre_percentage'] = mpg[1] / info['size']
+
+        return info
+
+    return mgp.Record(
+        communities=[get_community_info(c) for c in communities],
+        quality=quality
+    )
+```
+
+Now that we have our function in place let's test some algorithms. We will be
+checking out community detection using
+[greedy modularity maximization by Clauset-Newman-Moore](https://networkx.github.io/documentation/latest/reference/algorithms/generated/networkx.algorithms.community.modularity_max.greedy_modularity_communities.html#networkx.algorithms.community.modularity_max.greedy_modularity_communities)
+and
+[label propagation](https://networkx.github.io/documentation/latest/reference/algorithms/generated/networkx.algorithms.community.label_propagation.label_propagation_communities.html#networkx.algorithms.community.label_propagation.label_propagation_communities).
+
+```python
+@mgp.read_proc
+def greedy_modularity_communities(context: mgp.ProcCtx,
+                                  calculate_quality: bool=False
+                                 ) -> mgp.Record(communities=list, q
+                                                 quality=mgp.Any):
+    return _get_communities(
+        context,
+        nxa.community.greedy_modularity_communities,
+        calculate_quality
+    )
+
+@mgp.read_proc
+def label_propagation_communities(context: mgp.ProcCtx,
+                                  calculate_quality: bool=False
+                                 ) -> mgp.Record(communities=list,
+                                                 quality=mgp.Any):
+    return _get_communities(
+        context,
+        nxa.community.label_propagation_communities,
+        calculate_quality
+    )
+```
+
+In the above snippet, we can notice an optional argument `calculate_quality` and
+usage of the type `mgp.Any` which is provided by Memgraph.
+
+Let's see the results with:
+
+```opencypher
+CALL deezer_example.greedy_modularity_communities()
+YIELD communities
+UNWIND communities as community
+WITH community
+WHERE community.size > 10
+RETURN community.most_popular_genre, community.most_popular_genre_percentage, community.size
+ORDER BY community.size DESC;
+```
+
+and
+
+```opencypher
+CALL deezer_example.label_propagation_communities()
+YIELD communities
+UNWIND communities as community
+WITH community
+WHERE community.size > 10
+RETURN community.most_popular_genre, community.most_popular_genre_percentage, community.size
+ORDER BY community.size DESC;
+```
+
+Hmm, `Pop` sure is popular. Let's ignore that genre:
```

Comment:
I don't know if it missing, but as a quick reader I would like to see some results from query here.

Reply:
Didn't include it because all the other tutorials don't have any results. But considering the text, I should probably include it here.